File: app/src/main/java/com/example/test/ui/recipeB/RecipeViewModel.java
package com.example.test.ui.recipeB;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.test.ui.addRecipe.Recipe;

import java.util.ArrayList;
import java.util.List;

public class RecipeViewModel extends ViewModel {
    private final MutableLiveData<List<Recipe>> recipeBook;

    public RecipeViewModel() {
        recipeBook = new MutableLiveData<>(new ArrayList<>());

        // Initialize with some default recipes
        List<Recipe> initialRecipes = new ArrayList<>();
        initialRecipes.add(new Recipe("Pasta", List.of("Pasta", "Tomato Sauce", "Cheese"), "Boil the pasta and mix with sauce."));
        initialRecipes.add(new Recipe("Salad", List.of("Lettuce", "Tomato", "Cucumber", "Olive Oil"), "Chop all ingredients and mix."));

        recipeBook.setValue(initialRecipes);
    }

    public LiveData<List<Recipe>> getRecipeBook() {
        return recipeBook;
    }

    public void addRecipeBook(Recipe recipe) {
        List<Recipe> currentRecipes = recipeBook.getValue();
        if (currentRecipes != null) {
            // Create a new list to trigger LiveData update
            List<Recipe> updatedRecipes = new ArrayList<>(currentRecipes);
            updatedRecipes.add(recipe);
            recipeBook.setValue(updatedRecipes);
        }
    }
}

File: app/src/main/java/com/example/chewsyui/ui/recipeB/RecipeViewModel.kt
package com.example.chewsyui.ui.recipeB

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.chewsyui.ui.addRecipe.Recipe

class RecipeViewModel : ViewModel() {

    private val _recipeBook = MutableLiveData<List<Recipe>>()
    val recipeBook: LiveData<List<Recipe>> = _recipeBook

    init {
        val initialRecipes = mutableListOf(
            Recipe(
                "Pasta",
                listOf("Pasta", "Tomato Sauce", "Cheese"),
                "Boil the pasta and mix with sauce."
            ),
            Recipe(
                "Salad",
                listOf("Lettuce", "Tomato", "Cucumber", "Olive Oil"),
                "Chop all ingredients and mix."
            )
        )
        _recipeBook.value = initialRecipes
    }

    // Function to add a recipe
    fun addRecipe(recipe: Recipe) {
        val currentRecipes = _recipeBook.value?.toMutableList() ?: mutableListOf()
        currentRecipes.add(recipe)
        _recipeBook.value = currentRecipes
    }

    // Function to edit a recipe (update an existing recipe)
    fun editRecipe(updatedRecipe: Recipe) {
        val currentRecipes = _recipeBook.value?.toMutableList() ?: mutableListOf()
        val index = currentRecipes.indexOfFirst { it.recipeName == updatedRecipe.recipeName }
        if (index != -1) {
            currentRecipes[index] = updatedRecipe
            _recipeBook.value = currentRecipes
        }
    }

    // Function to remove a recipe
    fun removeRecipe(recipeToRemove: Recipe) {
        val currentRecipes = _recipeBook.value?.toMutableList() ?: mutableListOf()
        currentRecipes.remove(recipeToRemove)
        _recipeBook.value = currentRecipes
    }
}

